backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs

```diff
@@ -21,13 +23,52 @@ public async Task<ICollection<AvailabilitySlotsGroupByTeamsDto>> GetAllAvailabil
                     .ThenInclude(x => x.Location)
                 .Include(x => x.AvailabilitySlots)
                     .ThenInclude(x => x.Author)
-                .Where(x => x.AvailabilitySlots.Any(x => x.Type == SlotType.Team))
-                .Where(x => x.AvailabilitySlots.Any(x => x.Members.Any(x => x.UserId == id)))
+                .Where(x => x.AvailabilitySlots.Any(x => x.Type == SlotType.Team) && x.AvailabilitySlots.Any(x => x.Members.Any(x => x.UserId == id)))
+                .Select(x =>
+                    new AvailabilitySlotsGroupByTeamsDto
+                    {
+                        Name = x.Name,
+                        PageLink = x.PageLink,
+                        AvailabilitySlots = x.AvailabilitySlots
+                            .Select(y =>
+                                new AvailabilitySlotDto
+                                {
+                                    Id = y.Id,
+                                    Name = y.Name,
+                                    Link = y.Link,
+                                    Type = y.Type,
+                                    Size = y.Size,
+                                    IsEnabled = y.IsEnabled,
+                                    Author =
+                                        new UserDto
+                                        {
+                                            Id = y.Author.Id,
+                                            UserName = y.Author.Name,
+                                            Email = y.Author.Email,
+                                            Image = y.Author.ImagePath,
+                                            Phone = y.Author.PhoneNumber
+                                        },
+                                    Location = new LocationDto
+                                    {
+                                        Name = y.Location.Name,
+                                    },
+                                    Members = y.Members
+                                        .Select(m =>
+                                         new UserDto
+                                         {
+                                             Id = m.User.Id,
+                                             UserName = m.User.Name,
+                                             Email = m.User.Email,
+                                             Image = m.User.ImagePath,
+                                             Phone = m.User.PhoneNumber
+                                         })
+                                        .ToList()
+                                })
+                            .ToList()
+                    })
```

Comment:
Are you really need all this field?
You only need to load data that you will use on your page

Reply:
Yes, I need these data, I can remove user email and phone number of user (maybe also link in availability slot for now, but it will be used later, so I decided to leave it).
But other fields are displaying on availability page.
![image](https://user-images.githubusercontent.com/98499085/184328118-32d53f5c-52b2-4617-8dcb-1fd93407187e.png)